releasenotes/notes/deprecate-extensions-d7ab3fca19962e2e.yaml

```diff
@@ -0,0 +1,38 @@
+---
+deprecations:
+  - |
+    Progress towards deprecating the :mod:`qiskit.extensions` module to streamline the structure
+    of Qiskit. The following objects have been moved to :mod:`qiskit.circuit.library`
+    and will be deprecated in :mod:`qiskit.extensions` in a future release,
+
+    * :class:`~.library.HamiltonianGateGate`,
+    * :class:`~.library.Initialize`,
+    * :class:`~.library.Isometry`,
+    * :class:`~.library.generalized_gates.mcg_up_diag.MCGupDiag`,
+    * :class:`~.library.UCGate`,
+    * :class:`~.library.UCPauliRotGate`,
+    * :class:`~.library.UCRXGate`,
+    * :class:`~.library.UCRYGate`,
+    * :class:`~.library.UCRZGate`,
+    * :class:`~.library.UnitaryGate`,
```

Comment:
`DiagonalGate` is missing in this list, as well as the circuit method `diagonal`.

Reply:
Added in [80ace24](https://github.com/Qiskit/qiskit/pull/10725/commits/80ace241f0ceb7026aae493c8ac7e94cbfeb058b) and e2e7808